test/helper.lua

```diff
@@ -565,4 +699,65 @@ function helpers.skip_old_tarantool_cartridge_hotreload()
         "Cartridge hotreload tests stuck for vshard 0.1.22+ on Tarantool 2.2, 2.3 and 2.4")
 end
 
+function helpers.start_default_cluster(g, srv_name)
+    local cartridge_cfg = {
+        datadir = fio.tempdir(),
+        server_command = helpers.entrypoint_cartridge(srv_name),
+        use_vshard = true,
+        replicasets = helpers.get_test_cartridge_replicasets(),
+    }
+    local vshard_cfg = {
+        sharding = helpers.get_test_vshard_sharding(),
+        bucket_count = 3000,
+        storage_init = entrypoint_vshard(srv_name, 'storage_init', false),
+        router_init = entrypoint_vshard(srv_name, 'router_init', false),
+        all_init = entrypoint_vshard(srv_name, 'all_init', false),
+        crud_init = true,
+    }
+
+    helpers.start_cluster(g, cartridge_cfg, vshard_cfg)
+end
+
+function helpers.start_cluster(g, cartridge_cfg, vshard_cfg)
+    if g.params.backend == helpers.backend.CARTRIDGE then
+        helpers.skip_cartridge_unsupported()
+
+        local cfg = table.deepcopy(cartridge_cfg)
+        cfg.env = {
+            ['ENGINE'] = g.params.engine
+        }
+
+        g.cluster = helpers.Cluster:new(cfg)
+        g.cluster:start()
+    elseif g.params.backend == helpers.backend.VSHARD then
+        local cfg = table.deepcopy(vshard_cfg)
+        cfg.engine = g.params.engine
+
+        local global_cfg = vtest.config_new(cfg)
+        vtest.cluster_new(g, global_cfg)
+    end
+end
+
+function helpers.stop_cluster(cluster, backend)
+    if backend == helpers.backend.CARTRIDGE then
+        helpers.stop_cartridge_cluster(cluster)
+    elseif backend == helpers.backend.VSHARD then
+        cluster:drop()
+    end
+end
+
+function helpers.backend_matrix(base_matrix)
```

Comment:
https://github.com/tarantool/luatest/blob/18859f6a67d7f92bda12297869d56928d9d7dc88/luatest/helpers.lua#L91

may be useful

Reply:
Yes, it works in some cases (just backend + engine). But in a text with a large number of options it becomes non-universal.
Therefore, I decided to use another logic (implemented in the `backend_matrix`), but the same for almost tests.